lte/gateway/c/core/oai/tasks/grpc_service/SpgwServiceImpl.cpp

```diff
@@ -294,30 +294,42 @@ bool SpgwServiceImpl::fillUpPacketFilterContents(
 // ANY and it is equivalent to 0.0.0.0/0
 // But this function is called only for non-empty ipv4 string
 bool SpgwServiceImpl::fillIpv4(packet_filter_contents_t* pf_content,
-                               const std::string ipv4addr) {
-  const auto cidrNetworkExpect = folly::IPAddress::tryCreateNetwork(ipv4addr);
-  if (cidrNetworkExpect.hasError()) {
-    OAILOG_ERROR(LOG_UTIL, "Invalid address string %s \n", ipv4addr.c_str());
+                               const std::string ipv4network) {
+  const int slash_pos = ipv4network.find("/");
+  std::string ipv4addr = (slash_pos != std::string::npos)
+                             ? ipv4network.substr(0, slash_pos)
+                             : ipv4network;
+  in_addr addr;
```

Comment:
can we pull out all the actual string conversion? or formatting? functionality into a testable function? I would love some unit test coverage on this :)

Reply:
Pushed. Unit test for the new function will be added.

Reply:
Unit test has been added.